var/spack/repos/builtin/packages/rocblas/package.py

```diff
@@ -106,6 +106,7 @@ class Rocblas(CMakePackage):
     conflicts("amdgpu_target=gfx1011", when="@:4.2.1")
     conflicts("amdgpu_target=gfx1012", when="@:4.2.1")
     conflicts("amdgpu_target=gfx1030", when="@:4.2.1")
+    conflicts("%gcc@12.2.0", when="@5.2.1:5.2.3")
```

Comment:
To add more context, compiling libstdc++ version 12's `<memory>` header with `hipcc` is what fails to compile. In that sense this could be a blanket conflict between GCC 12 and HIP <= 5.2. However, it seems like rocBLAS doesn't include `<memory>` in every version (at least in the files that are compiled by `hipcc`) so there's also a lower bound on the version in the conflict.

Other projects that attempt to compile `<memory>` with `hipcc` and GCC 12 are likely to hit the same issue though.